editor.planx.uk/src/pages/Pay/MakePayment.tsx

```diff
@@ -44,137 +50,189 @@ enum PaymentState {
 export default function MakePayment({
   sessionPreviewData,
   createdAt,
-  paymentRequestId,
+  id: paymentRequestId,
+  govPayPaymentId,
   paymentAmount,
+  paidAt,
 }: PaymentRequest) {
+  const { address, rawProjectTypes } =
+    parseSessionPreviewData(sessionPreviewData);
   const [currentState, setState] = useState<
     (typeof States)[keyof typeof States]
   >(States.Init);
-  const [loading, isLoading] = useState(true);
-  const [payment, setPayment] = useState(
-    sessionPreviewData.govUkPayment || undefined
-  );
+  const [isLoading, setIsLoading] = useState(true);
+  const [payment, setPayment] = useState<GovUKPayment | undefined>(undefined);
+  const flowName = useStore((state) => state.flowName);
+  const theme = useTheme();
 
+  // Pass async errors up to ErrorBoundary
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
   useEffect(() => {
-    const updatePaymentState = async () => {
-      setState(States.Fetching);
-      fetchPayment({
-        paymentRequestId,
-        payment: sessionPreviewData.govUkPayment,
-      }).then((responseData: GovUKPayment | null) => {
-        if (responseData) resolvePaymentResponse(responseData);
-        isLoading(false);
-        switch (computePaymentState(responseData)) {
-          case PaymentState.NotStarted:
-            setState(States.Ready);
-            break;
-          case PaymentState.Pending:
-            setState(States.ReadyToRetry);
-            break;
-          case PaymentState.Failed:
-            setState(States.ReadyToRetry);
-            setPayment(undefined);
-            break;
-          case PaymentState.Completed:
-            setState(States.Finished);
-            handleSuccess();
-            break;
-        }
-      });
-    };
+    if (errorMessage) throw Error(errorMessage);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    // If payment is completed, we don't need to fetch data from GovPay
```

Comment:
Also - it would fail. We would throw an error when trying to update the session and paymentRequest a second time as no `affected_rows` would be returned.